src/components/ContactCard/ContactGroups.jsx

```diff
@@ -15,17 +16,17 @@ import SpinnerContact from '../Components/Spinner'
 import { checkIfGroupAlreadyExists } from '../ContactGroups/helpers/groups'
 import container from './ContactGroupsContainer'
 
-class ContactGroupsClass extends React.Component {
-  updateContactGroups = groups => {
+export class ContactGroupsClass extends React.Component {
+  updateContactGroups = nextGroups => {
     const { contact } = this.props
 
-    contact.groups.data.map(group => {
-      contact.groups.removeById(group._id)
-    })
-    groups.map(groupToAdd => {
-      contact.groups.addById(groupToAdd._id)
-    })
-    this.props.updateContact(contact)
+    const currentGroups = contact.groups.data
+    const toAdd = differenceBy(nextGroups, currentGroups, '_id')
+    const toRemove = differenceBy(currentGroups, nextGroups, '_id')
+
+    if (toAdd.length > 0) contact.groups.addById(toAdd.map(({ _id }) => _id))
+    else if (toRemove.length > 0)
+      contact.groups.removeById(toRemove.map(({ _id }) => _id))
```

Comment:
you can have toAdd and toRemove at the same time, no?

Reply:
And maybe we should add a test with this case : 
- adding 1 group 
- removing one other

Reply:
No we can't have that as far as I can tell — and actually it will break cozy-client if that happens. I still have to file a bug about that, but it's not one that can be fixed easily…

Reply:
Added a comment highlighting why we can't do this right now 👍